### PR TITLE
Fix variant::array

### PR DESCRIPTION
**Pull request by majetideepak**
Verify all array elements have the same TypeKind

### CODE REVIEW

**Review by mbasmanova — APPROVED** [review]
Thanks.

**Comment by mbasmanova** [comment]
@majetideepak There are CI failures. Would you take a look?

File: velox/type/Variant.h [hunk 1]
```diff
@@ -274,14 +274,28 @@ class variant {
...
   }
 
+  static inline void verifyArrayElements(const std::vector<variant>& inputs) {
+    if (!inputs.empty()) {
+      const auto elementTypeKind = inputs[0].kind();
+      for (int i = 1; i < inputs.size(); ++i) {
+        VELOX_CHECK_EQ(
+            elementTypeKind,
+            inputs[i].kind(),
+            "All array elements must be of the same kind");
```
[the remark this answers is not in the document]
Comment (majetideepak) [reply velox/type/Variant.h:284]: Maps can have different key vs. value types right? What check are you suggesting?

**Comment by majetideepak** [comment]
@mbasmanova the error you posted `(INTEGER vs. BIGINT)`  is a valid error. Do you see a lot of these?

**Comment by majetideepak** [comment]
These are valid errors. They have to be fixed.

**Comment by majetideepak** [comment]
@mbasmanova Luckily, the CI tests passed without any issues.

**Comment by kgpai** [comment]
I am working on merging this. @majetideepak

**Comment by majetideepak** [comment]
@kgpai thank you for the update!

**Pull request closed without merging**